test/test_ops.py

```diff
@@ -533,8 +535,13 @@ def test_noncontiguous_samples(self, device, dtype, op):
     #   incorrectly sized out parameter warning properly yet
     # Cases test here:
     #   - out= with the correct dtype and device, but the wrong shape
-    @ops(_ops_and_refs, dtypes=OpDTypes.none)
+    @ops(op_db, dtypes=OpDTypes.none)
+    @setLinalgBackendsToDefaultFinally
     def test_out_warning(self, device, op):
+        if torch.version.hip is not None:
+            if op.name == "linalg.svdvals":
+                torch.backends.cuda.preferred_linalg_library('magma')
```

Comment:
Instead of putting this condition in a generic test, the same effect would be achieved if `torch.backends.cuda.preferred_linalg_library('magma')` was put in `sample_inputs_linalg_svdvals` function in `torch/testing/_internal/opinfo/definitions/linalg.py`

Reply:
That being said, it may be better to simply have USE_CUSOLVER / CUDART_VERSION be set to false in these builds. Then, as you mention in the review, everything should go through the right path (and if it doesn't, it's a bug)

Reply:
Maybe a `PreferMagmaOnHIP` decorator for tests? Then it could be applied to specific tests and specific operators, and the intent of the decorator would be reasonably clear from the name. Plus the decorator itself could describe why it exists and how it's intended to be used